### ISSUE TITLE
Finding features with OpenMS or XCMS

### ISSUE THREAD

**Issue by Moo95** [rickhelmus/patRoon#15]
Hello Rick,

i am new in patroon and wanted to use the workflow. So i have completed the installation and can also run the example data but if i try to find features in my data it always gives me this notification. The data was a full scan with ddMS2 acquisition. Is this issue a problem with administrator privileges on the PC.

**With OpenMS**

fList <- findFeatures(anaInfo, "openms")
Finding features with OpenMS for 2 analyses ...
|   0%Fehler: Can not open file C:/Users/User/Documents/analyses/raw/SMX.mzML! Original error was: Error in pwizModule$open(filename): Invalid cvParam accession "1003112"

**With XCMS**

Feature statistics:
SMX: 0 (0.0%)
solvent-1: 35 (100.0%)
Total: 35
Warnmeldungen:
1: In serialize(data, node$con) :
  'package:stats' evtl. nicht verfügbar während des Ladens
2: In serialize(data, node$con) :
  'package:stats' evtl. nicht verfügbar während des Ladens
3: In xcmsSet(files, analysisInfo$analysis, analysisInfo$group, method = method,  :
  Peak detection failed in 'C:/Users/User/Documents/analyses/raw/SMX.mzML':Error: Can not open file C:/Users/User/Documents/analyses/raw/SMX.mzML! Original error was: Error in pwizModule$open(filename): Invalid cvParam accession "1003112"

Greetings

**Comment by rickhelmus** [comment]
Hello,

It seems that maybe something strange happened with the file conversion. How did you convert your files to mML?

**Comment by Moo95** [comment]
**Hi.

so i converted them with MSConvert. If i let patRoon convert the files it also happends:** 
-------

> anaInfo <- generateAnalysisInfo(paths = c("C:/Users/User/Documents/analyses/mzml"),
+                                 groups = c("Ozon-MeOH", "Ozon-RTI-10ng", "Ozon-RTI-1ng", "Ozon-RTI-20ng", "Ozon-RTI-5ng", "Ozon-RTI-Blank", "Ozon-RTI", "SMX-0-min-90"),
+                                 blanks = c("", "", "", "", "", "", "", ""))
> doDataPretreatment <- TRUE
> if (doDataPretreatment)
+ {
+     convertMSFiles(anaInfo = anaInfo, from = "thermo",
+                    to = "mzML", algorithm = "pwiz", centroid = "vendor")
+ } 
  |================================================================================================================| 100%
> # Find all features.
> # NOTE: see manual for many more options
> fList <- findFeatures(anaInfo, "openms")
**Finding features with OpenMS for 8 analyses ...
  |==============|  12%Fehler: Can not open file C:/Users/User/Documents/analyses/mzml/Ozon-RTI-5ng.mzML! Original error was: Error in pwizModule$open(filename): Invalid cvParam accession "1003112"**


**If i convert them at my own and load them into R than it works.**

 _> # Find all features.
> # NOTE: see XCMS manual for many more options
> fList <- findFeatures(anaInfo, "xcms3", param = xcms::CentWaveParam())
Finding features with XCMS for 8 analyses ...
Loading raw data...


Attache Paket: ‘BiocGenerics’

The following objects are masked from ‘package:parallel’:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ, clusterExport, clusterMap, parApply, parCapply, parLapply, parLapplyLB, parRapply,
    parSapply, parSapplyLB

The following objects are masked from ‘package:stats’:

    IQR, mad, sd, var, xtabs

The following objects are masked from ‘package:base’:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep, grepl,
    intersect, is.unsorted, lapply, Map, mapply, match, mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce, rownames,
    sapply, setdiff, sort, table, tapply, union, unique, unsplit, which.max, which.min

Welcome to Bioconductor

    Vignettes contain introductory material; view with 'browseVignettes()'. To cite Bioconductor, see 'citation("Biobase")', and for packages
    'citation("pkgname")'.


Attache Paket: ‘S4Vectors’

The following object is masked from ‘package:base’:

    expand.grid


Attache Paket: ‘ProtGenerics’

The following object is masked from ‘package:stats’:

    smooth


This is MSnbase version 2.15.7 
  Visit https://lgatto.github.io/MSnbase/ to get started.


Attache Paket: ‘MSnbase’

The following object is masked from ‘package:base’:

    trimws


This is xcms version 3.11.7 


Attache Paket: ‘xcms’

The following object is masked from ‘package:stats’:

    sigma

Detecting mass traces at 25 ppm ... OK
Detecting chromatographic peaks in 8121 regions of interest ... OK: 1950 found.
Detecting mass traces at 25 ppm ... OK
Detecting chromatographic peaks in 2489 regions of interest ... OK: 433 found.

Attache Paket: ‘BiocGenerics’

The following objects are masked from ‘package:parallel’:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ, clusterExport, clusterMap, parApply, parCapply, parLapply, parLapplyLB, parRapply,
    parSapply, parSapplyLB

The following objects are masked from ‘package:stats’:

    IQR, mad, sd, var, xtabs

The following objects are masked from ‘package:base’:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep, grepl,
    intersect, is.unsorted, lapply, Map, mapply, match, mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce, rownames,
    sapply, setdiff, sort, table, tapply, union, unique, unsplit, which.max, which.min

Welcome to Bioconductor

    Vignettes contain introductory material; view with 'browseVignettes()'. To cite Bioconductor, see 'citation("Biobase")', and for packages
    'citation("pkgname")'.


Attache Paket: ‘S4Vectors’

The following object is masked from ‘package:base’:

    expand.grid


Attache Paket: ‘ProtGenerics’

The following object is masked from ‘package:stats’:

    smooth


This is MSnbase version 2.15.7 
  Visit https://lgatto.github.io/MSnbase/ to get started.


Attache Paket: ‘MSnbase’

The following object is masked from ‘package:base’:

    trimws


This is xcms version 3.11.7 


Attache Paket: ‘xcms’

The following object is masked from ‘package:stats’:

    sigma

Detecting mass traces at 25 ppm ... OK
Detecting chromatographic peaks in 10446 regions of interest ... OK: 1543 found.
Detecting mass traces at 25 ppm ... OK
Detecting chromatographic peaks in 8962 regions of interest ... OK: 1747 found.

Attache Paket: ‘BiocGenerics’

The following objects are masked from ‘package:parallel’:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ, clusterExport, clusterMap, parApply, parCapply, parLapply, parLapplyLB, parRapply,
    parSapply, parSapplyLB

The following objects are masked from ‘package:stats’:

    IQR, mad, sd, var, xtabs

The following objects are masked from ‘package:base’:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep, grepl,
    intersect, is.unsorted, lapply, Map, mapply, match, mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce, rownames,
    sapply, setdiff, sort, table, tapply, union, unique, unsplit, which.max, which.min

Welcome to Bioconductor

    Vignettes contain introductory material; view with 'browseVignettes()'. To cite Bioconductor, see 'citation("Biobase")', and for packages
    'citation("pkgname")'.


Attache Paket: ‘S4Vectors’

The following object is masked from ‘package:base’:

    expand.grid


Attache Paket: ‘ProtGenerics’

The following object is masked from ‘package:stats’:

    smooth


This is MSnbase version 2.15.7 
  Visit https://lgatto.github.io/MSnbase/ to get started.


Attache Paket: ‘MSnbase’

The following object is masked from ‘package:base’:

    trimws


This is xcms version 3.11.7 


Attache Paket: ‘xcms’

The following object is masked from ‘package:stats’:

    sigma

Detecting mass traces at 25 ppm ... OK
Detecting chromatographic peaks in 9035 regions of interest ... OK: 1687 found.
Detecting mass traces at 25 ppm ... OK
Detecting chromatographic peaks in 9439 regions of interest ... OK: 1819 found.

Attache Paket: ‘BiocGenerics’

The following objects are masked from ‘package:parallel’:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ, clusterExport, clusterMap, parApply, parCapply, parLapply, parLapplyLB, parRapply,
    parSapply, parSapplyLB

The following objects are masked from ‘package:stats’:

    IQR, mad, sd, var, xtabs

The following objects are masked from ‘package:base’:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep, grepl,
    intersect, is.unsorted, lapply, Map, mapply, match, mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce, rownames,
    sapply, setdiff, sort, table, tapply, union, unique, unsplit, which.max, which.min

Welcome to Bioconductor

    Vignettes contain introductory material; view with 'browseVignettes()'. To cite Bioconductor, see 'citation("Biobase")', and for packages
    'citation("pkgname")'.


Attache Paket: ‘S4Vectors’

The following object is masked from ‘package:base’:

    expand.grid


Attache Paket: ‘ProtGenerics’

The following object is masked from ‘package:stats’:

    smooth


This is MSnbase version 2.15.7 
  Visit https://lgatto.github.io/MSnbase/ to get started.


Attache Paket: ‘MSnbase’

The following object is masked from ‘package:base’:

    trimws


This is xcms version 3.11.7 


Attache Paket: ‘xcms’

The following object is masked from ‘package:stats’:

    sigma


Detecting mass traces at 25 ppm ... OK
Detecting chromatographic peaks in 8936 regions of interest ... OK: 1740 found.
Detecting mass traces at 25 ppm ... OK
Detecting chromatographic peaks in 11672 regions of interest ... OK: 2755 found.
Done!


Feature statistics:
Ozon-MeOH: 1543 (11.3%)
Ozon-RTI-10ng: 1747 (12.8%)
Ozon-RTI-1ng: 1687 (12.3%)
Ozon-RTI-20ng: 1819 (13.3%)
Ozon-RTI-5ng: 1740 (12.7%)
Ozon-RTI-Blank: 2755 (20.1%)
Ozon-RTI: 1950 (14.3%)
SMX-0-min-90: 433 (3.2%)
Total: 13674
Warnmeldungen:
1: In serialize(data, node$con) :
  'package:stats' evtl. nicht verfügbar während des Ladens
2: In serialize(data, node$con) :
  'package:stats' evtl. nicht verfügbar während des Ladens
3: In serialize(data, node$con) :
  'package:stats' evtl. nicht verfügbar während des Ladens
4: In serialize(data, node$con) :
  'package:stats' evtl. nicht verfügbar während des Ladens_

**Comment by rickhelmus** [comment]
Hello!

I edited your post a bit to make it readable ;-)

Can you try to update ProteoWizard? I noticed today that I had a similar error when using it with patRoon, but not from the ProteoWizard GUI. This was very recently fixed.

**Comment by Moo95** [comment]
Hello,

sorry for that bad editing. I didnt really figured it out.
so u mean with updating, the newest version of Proteowizard? I installed the newest version. I just repeated the analysis and i tried it again and now it works. Thanks for ur help

> # Find all features.
> # NOTE: see manual for many more options
> fList <- findFeatures(anaInfo, "openms")
Finding features with OpenMS for 8 analyses ...
Done!
Feature statistics:
Ozon-MeOH: 19911 (13.5%)
Ozon-RTI-10ng: 18362 (12.5%)
Ozon-RTI-1ng: 17692 (12.0%)
Ozon-RTI-20ng: 18823 (12.8%)
Ozon-RTI-5ng: 17856 (12.1%)
Ozon-RTI-Blank: 26105 (17.7%)
Ozon-RTI: 18682 (12.7%)
SMX-0-min-90: 9912 (6.7%)
Total: 1473

So the issue can be closed.
